### ISSUE TITLE
Decide on signedness of various attributes

### ISSUE THREAD

**Issue by burmako** [openxla/stablehlo#481]
MHLO/StableHLO dialects frequently use `I64Attr` and `I64ElementsAttr` which are signless. HLO uses signed integers. In the spec, we're using signed integers. Let's review and make a consistent decision once and for all.

**Comment by ghpvnist** [comment]
To give an example, PR for ReducePrecisionOp #482 uses `I32Attr` for `exponent_bits` and `mantissa_bits`. The ODS enforces that these are `uint32`, but the spec currently says "constant of type `si32`".

Related to this, there may be inconsistencies between the use of "constant of type x" and "1-dimensional tensor constant of type x" for these attributes which may need to be sorted out.

**Comment by burmako** [comment]
I don't think that the ODS enforces anything. Since `I32Attr` means "signless", it just bitcasts the value to `uint32_t`. For example, this passes the verifier at the moment:

```mlir
func.func @main(%arg: tensor<f32>) -> tensor<f32> {
  %0 = "stablehlo.reduce_precision"(%arg) {
    exponent_bits = -5 : i32,
    mantissa_bits = 2 : i32
  } : (tensor<f32>) -> tensor<f32>
  func.return %0 : tensor<f32>
}
```

**Comment by GleasonK** [comment]
Signed integers in MLIR are very unused, I believe because they were added much later than signless integers. As such I don't think its a good idea to diverge from the MLIR standard way of representing signed values using the signless int type. Doing so would complicate all conversions to/from StableHLO, and make dialect interop much more difficult.

Closing this issue, if a reason to start considering this distinction comes up, or if upstream / MLIR community starts to use the type more, we can reconsider.
